src/Component/GeneralEntity/GeneralEntityRoot/GeneralEntityRoot.tsx

```diff
@@ -255,14 +258,42 @@ export function GeneralEntityRoot<T extends BaseEntity>({
    */
   useEffect(() => {
     if (id && id.toString() !== 'create' && id !== editEntity?.id) {
-      fetchEntity(parseInt(id.toString(), 10));
+      if (isPreviousFormDirty.current) {
+        modal.confirm({
+          title: t('GeneralEntityRoot.reminderModal.title'),
+          content: t('GeneralEntityRoot.reminderModal.description'),
+          okText: t('GeneralEntityRoot.reminderModal.accept'),
+          cancelText: t('GeneralEntityRoot.reminderModal.decline'),
+          onOk() {
+            onSaveClick();
+            fetchEntity(parseInt(id.toString(), 10));
+          },
+          onCancel() {
+            fetchEntity(parseInt(id.toString(), 10));
+            notification.info({
+              message: t('GeneralEntityRoot.saveWarning', {
+                entity: TranslationUtil.getTranslationFromConfig(entityName, i18n)
+              })
+            });
+          }
+        });
+        isPreviousFormDirty.current = formIsDirty;
+      }
+      else {
+        fetchEntity(parseInt(id.toString(), 10));
+      }
     }
     if (id && id.toString() === 'create' && editEntity === undefined) {
       const e: T = entityController?.createEntity();
       setEditEntity(e);
       form.setFieldsValue(entityController?.getEntity());
     }
-  }, [id, fetchEntity, editEntity, entityController, form]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id, fetchEntity, formIsDirty, editEntity, entityController, form, modal]);
+
+  useEffect(() => {
+    isPreviousFormDirty.current = formIsDirty;
+  }, [formIsDirty]);
```

Comment:
you are setting `isPreviousFormDirty.current` to the value of `formIsDirty` every time it is changed. Why do you need `isPreviousFormDirty`?

Reply:
This is used above as a condition to trigger the modal. I use this so when the form is changed (i.e name change or code change) The form is set to dirty and is saved to the useRef. Using formIsDirty does not do the job since when changing entities formIsDirty is set to the new entity, therefore it always returns false.

Reply:
so to understand the code here, this is what i think is happening:
1. you change something in the form
2. formIsDirty changes, both hooks run
3. `id === editEntity.id` so the first hook does nothing
4. in the second hook `isPreviousFormDirty.current` is set to `true`
5. You change the entity. I suppose the `id` changes and `formIsDirty` is set to `false`
6.  `id !== editEntity.id && isPreviousFormDirty.current` so the modal is called
7. in the second hook `isPreviousFormDirty.current` is set to `false`

The problem is that in step 6 and 7 you rely on the order of the hooks. That is a creative way for solving this, but quite unusual and might be hard to understand for future developers.

Let me think about it for a moment.

Reply:
You might be able to do something like this, but you would need to test this, if you would want to do it.

```typescript
const [isPreviousFormDirty, setIsPreviousFormDirty] = useState<boolean>(false);

useEffect(() => {
    if (id && id.toString() !== 'create' && id !== editEntity?.id) {
      if (isPreviousFormDirty) {
        modal.confirm({
          title: t('GeneralEntityRoot.reminderModal.title'),
          content: t('GeneralEntityRoot.reminderModal.description'),
          okText: t('GeneralEntityRoot.reminderModal.accept'),
          cancelText: t('GeneralEntityRoot.reminderModal.decline'),
          onOk() {
            onSaveClick();
            fetchEntity(parseInt(id.toString(), 10));
          },
          onCancel() {
            fetchEntity(parseInt(id.toString(), 10));
            notification.info({
              message: t('GeneralEntityRoot.saveWarning', {
                entity: TranslationUtil.getTranslationFromConfig(entityName, i18n)
              })
            });
          }
        });
      }
      else {
        fetchEntity(parseInt(id.toString(), 10));
      }
      setIsPreviousFormDirty(false); // here we either saved or discarded the changes, so we don't need the value anymore
    }
    if (id && id.toString() === 'create' && editEntity === undefined) {
      const e: T = entityController?.createEntity();
      setEditEntity(e);
      form.setFieldsValue(entityController?.getEntity());
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [id, fetchEntity, previousFormIsDirty, editEntity, entityController, form, modal]);

  useEffect(() => {
    if (formIsDirty) {
      setIsPreviousFormDirty(true); // only set if it changes from false to true, but not when it changes back
    }
  }, [formIsDirty]);
```

Reply:
I understand. I tested the code snippet and it works well for opening the modal. Yet there is one problem in my version the application waits for the confirmation of the modal. In the code snippet the entity changes despite the modal popping up therefore `onSaveClick()` does not save the entity that should be saved. I'm currently looking for a fix for it :+1: Thanks for the thorough suggestion :D

Reply:
can you `await` the `onSaveClick` function?

Reply:
there was no need for an await... there was one hook that triggered the function wrongly and therefore the function would jump to `fetchEntity` in the else condition. Now everything works as intended ^-^ thanks @simonseyock :D